### PR TITLE
fix: re-add the author onboarding

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Re-introducing the author onboarding
- For authors this adds the Twitter field to the signup

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-720 #done

### CODE REVIEW

**Review by idoshamun — APPROVED** [review]

File: packages/shared/src/components/auth/SocialRegistrationForm.tsx [hunk 1]
```diff
@@ -58,6 +66,8 @@ export const SocialRegistrationForm = ({
   const { user } = useContext(AuthContext);
   const [nameHint, setNameHint] = useState<string>(null);
   const [usernameHint, setUsernameHint] = useState<string>(null);
+  const [twitterHint, setTwitterHint] = useState<string>(null);
+  const isAuthorOnboarding = trigger === AuthTriggers.Author || true;
```
Comment (sshanzel) [inline packages/shared/src/components/auth/SocialRegistrationForm.tsx:70]: This will always be true.

Probably this was for dev mode.

**Review by sshanzel — CHANGES_REQUESTED** [review]
Thank you for collecting all the triggers to use an `enum`! It is so much cleaner now.

Just had one comment that was probably used for dev mode.

File: packages/shared/src/components/auth/SocialRegistrationForm.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/auth/SocialRegistrationForm.tsx:70]: Oh yes, indeed! sharp eyes 🔭

**Review by sshanzel — APPROVED** [review]
Nice fix! 🚢 🚢 🚢

File: packages/shared/src/components/auth/SocialRegistrationForm.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/auth/SocialRegistrationForm.tsx]: Happens to me always, so I know 🤣

**Pull request merged**